### PR TITLE
Golang

### PR DESCRIPTION
**Pull request by joshmarinacci**

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: GoLang Tests
```
Comment (EliSchleifer) [inline .github/workflows/go.yaml:1]: GoLang is spelled Golang

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go Action
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22.5'
+      - name: Install dependencies
+        run: |
+          go install github.com/jstemmer/go-junit-report/v2@latest
```
Comment (EliSchleifer) [inline .github/workflows/go.yaml:21]: I think idiomatic way is to use a go.mod file and install from that

module example.com/mymodule

go 1.14

require (
    example.com/othermodule v1.2.3
    example.com/thismodule v1.2.3
    example.com/thatmodule v1.2.3
)

replace example.com/thatmodule => ../thatmodule
exclude example.com/thismodule v1.3.0

File: go/go-junit-report/readme.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+
+# go 
+
+GoLang has a built in testing tool invoked with `go test`.
```
Comment (EliSchleifer) [inline go/go-junit-report/readme.md:4]: Golang

File: go/go-junit-report/readme.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+https://github.com/jstemmer/go-junit-report
+
+go-junit-report will convert the standard testing output into junit xml 
+
+go do the src dir
```
Comment (EliSchleifer) [inline go/go-junit-report/readme.md:10]: go do?

File: go/go-junit-report/readme.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+Install go
+
+* **Macos** `brew install go`
+
+make sure `$HOME/go/bin` is in your PATH
+
+Install go-junit-report
+
+```shell
+go install github.com/jstemmer/go-junit-report/v2@latest
```
Comment (EliSchleifer) [inline go/go-junit-report/readme.md:29]: I think you want
`go get {foo}`

and then you can do go build etc...

File: .github/workflows/go.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Run go-junit-report tests
+        uses: trunk-io/analytics-uploader@main
+        with:
+          # Path to your test results.
+          junit-paths: "**/*_test.xml"
+          # Provide your Trunk organization slug.
+          org-slug: trunk-staging-org
+          # Provide your Trunk API token as a GitHub secret.
+          token: ${{ secrets.TRUNK_STAGING_ORG_API_TOKEN }}
+          run: sh go/go-junit-report/runtests.sh
```
Comment (EliSchleifer) [inline .github/workflows/go.yaml:33]: I think this helper runtest is doing very little for us here. 

I think you can leave the readme.md files in place and just fix up the command lines to use the tools directly.

then the awkward `cd` commands can be removed

File: go/src/hello_test.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+
+
+// TestHelloName calls greetings.Hello with a name, checking
+// for a valid return value.
+func TestHelloName(t *testing.T) {
+    name := "Gladys"
+    want := regexp.MustCompile(`\b`+name+`\b`)
+    msg, err := Hello("Gladysx")
```
Comment (EliSchleifer) [inline go/src/hello_test.go:15]: What actually is this test doing. How does the random element above impact passing logic

File: .github/workflows/go.yaml (hunk 2 above)
Comment (joshmarinacci) [reply .github/workflows/go.yaml:21]: `go-junit-report` and `gotestsum ` are CLI commands. This was how the instructions said to install it.

File: go/go-junit-report/readme.md (hunk 5 above)
Comment (joshmarinacci) [reply go/go-junit-report/readme.md:29]: `install` will make it usable as a cli program.

File: .github/workflows/go.yaml (hunk 6 above)
Comment (joshmarinacci) [reply .github/workflows/go.yaml:33]: I tried doing that originally. Unfortunately the `go` commands kept getting run in the wrong directory.  Doing `run: |` with multiple commands as an argument to our uploader action didn't work either. It seemed to only run the first of the multiple commands. So I used the shell as a temp fix until we figure out a better solution.

File: go/src/hello_test.go (hunk 7 above)
Comment (joshmarinacci) [reply go/src/hello_test.go:15]: Right now this isn't random. I just wanted to have some test that I could upload results from. For each of our examples should it randomly fail? What % of the time?

Comment (EliSchleifer) [reply go/src/hello_test.go:15]: I guess I saw usage of random and expected some failure expectation - 

I think it's fine to have tests that pass - I just wasn't sure what expectation was of this function and what test was checking really.

File: .github/workflows/go.yaml (hunk 6 above)
Comment (EliSchleifer) [reply .github/workflows/go.yaml]: Did you file a bug that that run: is not behaving like the run does in GH action step - we might need to fix that

**Review by EliSchleifer — APPROVED** [review]
LGTM. Let's file tickets for the working-directory thing and the fact that run: doesn't behave as expected

**Pull request merged**